padguidedb/padguidedb.py

```diff
@@ -111,40 +118,74 @@ async def searchdungeon(self, ctx, *, search_text):
             for page in pagify(msg):
                 await ctx.send(box(page))
 
-    @padguidedb.command()
-    @checks.is_owner()
-    async def setdungeonscriptfile(self, ctx, *, dungeon_script_file):
-        """Set the dungeon script file."""
-        self.settings.setDungeonScriptFile(dungeon_script_file)
-        await ctx.tick()
+    @padguidedb.group()
+    async def series(self, ctx):
+        """Series"""
 
-    @padguidedb.command()
-    @checks.is_owner()
-    async def setfulletlfile(self, ctx, *, full_etl_file):
-        """Set the full ETL file."""
-        self.settings.setFullETLFile(full_etl_file)
-        await ctx.tick()
+    @series.command(name="search")
+    async def series_search(self, ctx, *, search_text):
+        """Search"""
+        search_text = '%{}%'.format(search_text)
+        with self.get_cursor() as cursor:
+            sql = ('select series_id, name_en, name_ja from series'
+                   ' where lower(name_en) like %s or lower(name_ja) like %s'
+                   ' order by series_id desc limit 20')
+            cursor.execute(sql, [search_text, search_text])
+            results = list(cursor.fetchall())
+            msg = 'Results\n' + json.dumps(results, indent=2, ensure_ascii=False)
+            await ctx.send(inline(sql))
+            for page in pagify(msg):
+                await ctx.send(box(page))
 
-    @padguidedb.command()
-    @checks.is_owner()
-    async def setimageupdatefile(self, ctx, *, image_update_file):
-        """Set the image update file."""
-        self.settings.setImageUpdateFile(image_update_file)
-        await ctx.tick()
+    @series.command(name="add")
+    async def series_add(self, ctx, *, elements: TokenConverter):
+        """Add"""
+        if not all(x in SERIES_KEYS for x in elements):
+            await ctx.send("Invalid key.  Valid keys are: `{}`".format("` `".join(SERIES_KEYS)))
+            return
 
-    @padguidedb.command()
-    @checks.is_owner()
-    async def setpythonexecutable(self, ctx, *, python_executable):
-        """Set the python executable file."""
-        self.settings.setPythonExecutable(python_executable)
-        await ctx.tick()
+        EXTRAS = {}
+        with self.get_cursor() as cursor:
+            cursor.execute("SELECT MAX(series_id) FROM series")
+            max_val = cursor.fetchall()[0]['MAX(series_id)']
+            EXTRAS['series_id'] = max_val + 1
+        EXTRAS['tstamp'] = int(datetime.now().timestamp())
+        elements = {**SERIES_KEYS, **EXTRAS, **elements}
+        
+        key_infix = ", ".join(elements.keys())
+        value_infix = ", ".join("%s" for v in elements.values())
+        with self.get_cursor() as cursor:
```

Comment:
can you make a helper function to do sql writes? there seems to be a few duplicated lines between the `INSERT` `UPDATE` and `DELETE` commands. specifically: 

```
with self.get_cursor() as cursor:
...
await ctx.send(inline(cursor._executed))
await ctx.send("{} row(s) added.".format(affected))
```